src/pages/MyCoverCropList/MyCoverCropComparison/MyCoverCropComparisonTable.js

```diff
@@ -107,22 +111,33 @@ const MyCoverCropComparisonTable = () => {
       tempRows.push(newRow);
     });
   };
+  const tableRef = useRef(null);
 
   const buildTableHeaders = () => selectedCrops.map((crop, index) => (
     <TableCell key={index} align="center">
-      <CropCard
-        crop={crop}
-        index={index}
-        dispatchRedux={dispatchRedux}
-        handleModalOpen={handleModalOpen}
-      />
+      <Typography
+        variant="subtitle1"
+        sx={{
+          fontWeight: 'bold',
+          color: 'primary.main',
+          cursor: 'pointer',
+          textDecoration: 'none',
+          '&:hover': {
+            textDecoration: 'underline',
+            color: 'primary.dark',
+          },
+          transition: 'color 0.3s ease',
+        }}
+        onClick={() => handleModalOpen(crop)}
```

Comment:
I found that this onClick function is making a lots of same api calls which will freeze the page for a few seconds. Can you look into this and try to solve that?

Reply:
Just noticed that in this same function the `<CropDetailsModal>` is created for each crop so there'll be a lot of same modals created when `onCLick` triggers. Can you move the modal outside of this function, so there'll be only one modal popup when clicked?

Reply:
I've moved cropdetailsmodal outside but kept the onclick function. It'll just send the data that needs to be attached to the modal